### ISSUE TITLE
How to handle heterogenous object sizes with diameter parameter

### ISSUE THREAD

**Issue by smt29021** [MouseLand/cellpose#472]
Hi,

Thank you for your amazing work.

I'm using the pre-trained nuclei segmentation model on a dataset where the size and shape of the cells can differ quite substantially.

I'm noticing a significant performance drop  when these images with heterogenous nuclei are being used as inputs. You outline in your documentation the impact of a too small or too high diameter but I was wondering what you would advise when the approximated diameter is suitable for only some of the objects in the image. I can provide example images upon request.

Thank you in advance for your help!

Samuel

**Comment by kevinjohncutler** [comment]
Please do post an example image, I am curious what your size range is. I have found that resizing really isn't necessary when you train your own models (especially with Omnipose), so that might be your next step.

**Comment by smt29021** [comment]
At the moment I'm just using CellPose as a pre-trained nuclei segmentation model without any fine-tuning. 

![image](https://user-images.githubusercontent.com/62348542/158341378-3a87687f-939c-4fdb-addb-51b165c8133b.png)


The example set of four images above shows my input image at position [0,0] with variation in the size of the nuclei. Image at position [0,1] is the output from using:
_python -m cellpose --dir /dir/to/imgs --pretrained_model nuclei --chan[0,0] --diameter 0_
Based on your comments in https://github.com/MouseLand/cellpose/issues/454 I adjusted the transforms. py normalise99 range to be 0.01 - 99.99 I then reran the segmentation and produced the image at position [0,2]. Finally when I still wasn't getting the results I was expecting I ran with the cytoplasm model which across a collection of images produces much better results (example shown at position [0,3]). You'll see for each of the three masks I have also added the approximated diameter from CellPose. Additional examples below:

![image](https://user-images.githubusercontent.com/62348542/158343225-b61abfbb-6498-4e23-8574-25f6bc123c1c.png)

![image](https://user-images.githubusercontent.com/62348542/158343366-89593abd-5fec-4c9a-943b-3e5e41c8a289.png)

It may be that the size range (and its impact on the diameter est) may not be the issue. Wondering why the cytoplasm model performs better on DAPI channel that the nuclei stained model.

**Comment by smt29021** [comment]
Is this in part due to the rescaling? I observe the significant difference in the approximated diameter for Cytoplasm vs Nuclei models which is to be expected. From your documentation for the Cytoplasm model the numerator in the rescaling is 30/diameter compared to 17/diameter for the nuclei model.

**Comment by carsen-stringer** [comment]
if your nuclei are very non-uniform, then the cytoplasm model may perform better. the nuclei in the nucleus training set were fairly homogeneous. you may alternatively want to train your own model on your data, this is now possible in the GUI (`pip install cellpose --upgrade`)
